src/pages/components/controls/filenameControls.tsx

```diff
@@ -8,25 +10,61 @@ type PatternTokenProps = {
 }
 
 export const PatternToken = memo(
-  ({ tokenName, isOn, handleChange }: PatternTokenProps) => {
-    const [state, setState] = useBoolean(isOn)
+  ({ tokenName, isOn, handleChange }: PatternTokenProps) => (
+    <Box
+      pl={'1em'}
+      pr={'1em'}
+      fontSize={'1em'}
+      fontWeight={'medium'}
+      color={'gray.900'}
+      borderRadius={'full'}
+      bg={isOn ? 'token.active' : 'white'}
+      cursor="pointer"
+      onClick={() => handleChange(!isOn)}
+    >
+      {tokenName}
+    </Box>
+  )
+)
+
+type SortablePatternTokenProps = {
+  token: string
+  name: string
+  handleChange?: (state: boolean) => void
+}
+
+export const SortablePatternToken = memo(
+  ({ token, name, handleChange }: SortablePatternTokenProps) => {
+    const { attributes, listeners, setNodeRef, transform, transition } = useSortable({
+      id: token,
+    })
+
+    const isDragging = attributes['aria-pressed']
+
     return (
-      <Box
-        pl={'1em'}
-        pr={'1em'}
-        fontSize={'1em'}
-        fontWeight={'medium'}
-        color={'gray.900'}
-        borderRadius={'full'}
-        bg={state ? 'token.active' : 'white'}
-        cursor="pointer"
-        onClick={() => {
-          setState.toggle()
-          handleChange(!state)
-        }}
+      <Tag
+        ref={setNodeRef}
+        {...attributes}
+        {...listeners}
+        size="md"
+        transition={transition}
+        fontSize="1em"
+        fontWeight="medium"
+        borderRadius="full"
+        color="gray.900"
+        bg="white"
+        userSelect="none"
+        transform={
+          transform ? `translate3d(${transform.x}px, ${transform.y}px, 0)` : undefined
+        }
+        cursor={isDragging ? 'grabbing' : 'grab'}
+        opacity={isDragging ? 0.65 : 1}
+        zIndex={isDragging ? 1 : 0}
       >
-        {tokenName}
-      </Box>
+        <TagLeftIcon as={DragHandleIcon} boxSize={3} color={'gray.600'} />
+        <TagLabel>{name}</TagLabel>
+        <TagCloseButton onClick={() => handleChange(false)} />
```

Comment:
The close button didn't work properly.
It looks like `TagCloseButton` should not be wrapped under `listeners`.
But if using a div to wrap `TagLeftIcon` and `TagLabel` it will break the style of tag.